week7/PhotoInspiration/app/src/main/java/com/manuflowers/photoinspiration/ui/home/HomeViewModel.kt

```diff
@@ -0,0 +1,67 @@
+package com.manuflowers.photoinspiration.ui.home
+
+import android.app.Application
+import androidx.lifecycle.*
+import com.manuflowers.photoinspiration.R
+import com.manuflowers.photoinspiration.application.PhotoInspirationApplication
+import com.manuflowers.photoinspiration.data.PhotosInspirationRepository
+import com.manuflowers.photoinspiration.data.local.database.PhotoInspirationDatabase
+import com.manuflowers.photoinspiration.data.local.database.PhotosDao
+import com.manuflowers.photoinspiration.data.local.preferences.PhotoInspirationPreferences
+import com.manuflowers.photoinspiration.data.models.ErrorNetworkMessage
+import com.manuflowers.photoinspiration.data.models.PhotoEntity
+import com.manuflowers.photoinspiration.data.remote.networking.NetworkStatusChecker
+import com.manuflowers.photoinspiration.data.remote.networking.RemoteApiManager
+import com.manuflowers.photoinspiration.data.remote.networking.buildApiService
+import kotlinx.coroutines.launch
+
+class HomeViewModel(application: Application) : AndroidViewModel(application) {
+
+    private val apiService by lazy { buildApiService() }
+
+    private val remoteApi by lazy { RemoteApiManager(apiService) }
+
+    private val repository: PhotosInspirationRepository
+
+    init {
+        val photosDao: PhotosDao =
```

Comment:
You should pass these properties through the constructor and use ViewModel.Factory pattern instead of using an AndroidViewModel, and building these things in the `init` block.